pkg/sources/source.go

```diff
@@ -35,3 +38,96 @@ type Source interface {
 	RoutesNotExported(ctx context.Context, neighborID string) (*api.RoutesResponse, error)
 	AllRoutes(context.Context) (*api.RoutesResponse, error)
 }
+
+func hiddenNeighborsExcludeLists(hiddenNeighbors []string) ([]*net.IPNet, []net.IP, []*regexp.Regexp, error) {
+	excludeCIDRs := make([]*net.IPNet, 0, len(hiddenNeighbors))
+	excludeIPs := make([]net.IP, 0, len(hiddenNeighbors))
+	excludePatterns := make([]*regexp.Regexp, 0, len(hiddenNeighbors))
+	for _, hiddenNeighbor := range hiddenNeighbors {
+		if _, hiddenNet, err := net.ParseCIDR(hiddenNeighbor); err == nil {
+			excludeCIDRs = append(excludeCIDRs, hiddenNet)
+		} else if ip := net.ParseIP(hiddenNeighbor); ip != nil {
+			excludeIPs = append(excludeIPs, ip)
+		} else {
+			pattern, err := regexp.Compile(hiddenNeighbor)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			excludePatterns = append(excludePatterns, pattern)
+		}
+	}
+	return excludeCIDRs, excludeIPs, excludePatterns, nil
+}
+
+func FilterHiddenNeighbors(neighbors api.Neighbors, hiddenNeighbors []string) (api.Neighbors, error) {
```

Comment:
If the `hidden_members` config option is parsed into a e.g. []*NeighborFilter / filterset something like
```
func (filters []*NeighborFilter) Exclude[T NeighborFilterable](n []T) ([]T, error) {
  for filters ... {
     if n.Match(...) skip
  }
}
```

Then the invocation in the source could be like 
```
neighbors := make(api.Neighbors, 0)
... decode neighbors...
response.Neighbors, err = src.cfg.HiddenNeighbors.Exclude(neighbors)
	if err != nil {
		return nil, err
	}
```

Reply:
Or: even better: before it is appended during decode:

```
if cfg.HiddenNeighbors.Match(...) {
  continue
 }
 response.Neighbors = append(...
 ```

Then the Exclude function would not even be required. :)